drivers/dac/ad5770r/ad5770r.c

```diff
@@ -631,6 +631,34 @@ int32_t ad5770r_set_monitor_setup(struct ad5770r_dev *dev,
 	return ret;
 };
 
+/**
+ * @brief Deallocate memory for the GPIOs assigned.
+ * @param dev - Device driver handler.
+ * @return SUCCESS in case of success, FAILURE otherwise.
+ */
+static int32_t ad5770r_remove_gpio(struct ad5770r_dev *dev)
```

Comment:
I think this function is not needed. You can do this inside of ad5770r_remove

Reply:
I initially wanted to include more changes in a future PR because no `ret` is actually checked in the initialization function. This function will help avoid code duplication and memory leaks when I start checking for fails in the init function. But I will add that now as well.